### ISSUE TITLE
isAirTransportable = canBlitz

### ISSUE THREAD

**Issue by panguitch** [triplea-game/triplea#444]
<blockquote>Veqryn wrote:
It seems that any land unit with 2 or more movement, that is also either 'isInfantry' or 'isAirTransportable', automatically is not checked for blitzing. Probably because the original coder thought they were being land-mech-transported or air transported over those territories.
    <a href="http://tripleadev.1671093.n2.nabble.com/TripleA-1-6-1-4-Stable-has-been-released-tp7582324p7583107.html">http://tripleadev.1671093.n2.nabble.com/TripleA-1-6-1-4-Stable-has-been-released-tp7582324p7583107.html</a></blockquote>

I've run up against this myself: http://tripleadev.1671093.n2.nabble.com/isAirTransportable-canBlitz-td7587868.html

I'm wondering if there's any chance of this being fixed. It basically means that any air-transportable unit automatically can blitz through any territory, even if lacks canBlitz and isn't being airlifted.

Repro steps:

> Yes, it still exists. Just tested again in 1.9.0.0.1627 To recreate an example related to mechanized infantry:
> Edit the POS2 xml to give infantry and artillery movement=2
> Then launch a game and as Russia, edit out the German troops in West Russia and Belorussia.
> You will now see that not only can the armour blitz from Caucus over West Russia to Belorussia, but so can the infantry. But not the artillery. This is because of "isInfantry"
> To recreate an example related to air transportables:
> Go back into the POS2 xml and change Russia's techattachment paratroopers to "true". Both infantry and artillery are already isairtransportable
> Launch the game as above.
> Now not only can the infantry blitz, but the artillery can also.
> So any unit with 2+ movement can always blitz, if it has isinfantry, or if it has isairtransportable and that nation has the paratrooper tech. This is regardless of the canBlitz unit attachment, and regardless of whether terrain restrictions should prevent blitz (which can be seen by heroes blitzing all terrains in Greyhawk Wars).

**Comment by ron-murhammer** [comment]
@panguitch - Yeah again weird air transport bugs. These are really hard to fix in the current system as its sort of a disaster.

**Comment by frigoref** [comment]
I tried to address this issue, but the test case I have created works as expected:
https://github.com/frigoref/triplea/tree/issue444_canBlitz
Any hint, what I should test differently?

**Comment by panguitch** [comment]
Yes, still exists. How to recreate: http://tripleadev.1671093.n2.nabble.com/isAirTransportable-canBlitz-td7587868.html#a7592459

**Comment by frigoref** [comment]
A question to the rule gurus:
Can the ability to blitz with armors be gained during the game or is it a static property for a unit type? Is it always a property for all players or can there be users with and users without this property?
My understanding is it is static and can be set for the respective unitAttachment with the sub-tag entry
<option name="receivesAbilityWhenWith" value="canBlitz:armour"/>

The new tests I added are working as expected above. Hence there would be only a small change needed which currently avoids the canBlitz-check for units with the property isInfantry or isAirTransportable. Otherwise, I can we have to rethink the current design for the canBlitz property.

**Comment by panguitch** [comment]
Definitely not a rules guru. But as I understand it (and copying from the POS2 xml) in addition to unitAttachment "canBlitz" and territoryEffectAttachment "noBlitz" there is this unitAttachment:

receivesAbilityWhenWith				values: gives a unit an ability when they are with, or on the same route as, another unit. currently only works with 'canBlitz'. example: value="canBlitz:armour"

These are static and unit-specific, not player-specific. However, there is the techAbilityAttachment "unitAbilitiesGained", which as per POS2 "allows unit types to gain abilities through this tech.  must be the name of the unit, followed by a colon ":", followed by all the abilities gained (separated by colons).  currently allows only "canBlitz" and "canBombard".  Effects from multiple techs are added together."

I've never used that myself, so don't know exactly how it works.

In any case, units being transported should be able to pass through multiple territories if their transport can. The problem is that this was achieved by making it so that transportable units don't get checked for blitz, regardless of whether they're being transported or not.

**Comment by frigoref** [comment]
So there is no way that in the same game one player's e.g. infantry has the ability to blitz with an armour and another player's units of the same type would not have that ability?

**Comment by panguitch** [comment]
One player's infantry might have "canBlitz" via the "unitAbilitiesGained" tech, but I don't see any way to make "receivesAbilityWhenWith" vary from player to player.

**Comment by frigoref** [comment]
There are currently some mixed up tests and meanings of the canBlitz property, so let me summarize what I assume and please let me know if there is anything wrong or if you fully agree. For clarity I provide reference links to http://axisandallies.wikia.com
After we clarified what should be happening I will rework the code accordingly.

[Definition](http://axisandallies.wikia.com/wiki/CanBlitz): "This unit property [**canBlitz**] allows a unit to move through an unoccupied enemy-controlled land area without stopping, capturing it along the way."
There is one exception for map properties [WW2V2](http://axisandallies.wikia.com/wiki/WW2V2) and [Blitz Through Factories And AA Restricted](http://axisandallies.wikia.com/wiki/Blitz_Through_Factories_And_AA_Restricted), i.e., [Factories](http://axisandallies.wikia.com/wiki/Factory) and [AA Gun](http://axisandallies.wikia.com/wiki/AA_Gun) prevent blitzing through their territories


From this definition and the games I know so far  I deduct that there are only 2 cases where units have the canBlitz property:
- [armour units](http://axisandallies.wikia.com/wiki/Armor) by default for all players
- [mechanized infantry units](http://axisandallies.wikia.com/wiki/Mechanized_Infantry) by default for all players

Therefore blitzing has nothing to do with 
- [paratrooper rule](http://axisandallies.wikia.com/wiki/Paratroopers)(Bombers can carry 1 infantry unit, for combat or non-combat movement. If moved into combat, the bomber does not participate, but can be shot down by AA) and 
- [mechanizedInfantry tech](http://axisandallies.wikia.com/wiki/MechanizedInfantry)(allows isLandTransport units to carry isInfantry units) 

which are about how to move. In neither of those cases is it possible to capture "an unoccupied enemy-controlled land area" along the way. More precisely, either the transported units are dropped in this land to conquer it or they are moved over it which would leave the ownership to the enemy.

What I am currently not clear on is: Should it be possible to have a unit without canBlitz property to cross "an unoccupied enemy-controlled land area" or should it have to stop? My first assumption was it has to stop, but then how about air units?

**Comment by panguitch** [comment]
Well, non-WW2 maps will use canBlitz differently. For instance, in my Greyhawk Wars map every unit has at least movement 2, and almost every unit has canBlitz, it's just restricted by various territory effects.

But yes, blitzing should have nothing to do with paratroopers (since no land unit is ever actually placed in the intervening territory, just transported above and over it). And it should have nothing to do directly with mechanisedinfantry. The transporting unit may or may not have blitz, but the unit being transported should be irrelevant to whether blitzing is happening.

Units without canBlitz cannot move through an unoccupied territory, regardless of how much movement they have left. They must stop and conquer. This should be irrelevant for air units since they're never actually in the territories they fly over.

**Comment by DanVanAtta** [comment]
I could not reproduce as described here in the source forge thread:

```
Yes, it still exists. Just tested again in 1.9.0.0.1627 To recreate an example related to mechanized infantry:
Edit the POS2 xml to give infantry and artillery movement=2
Then launch a game and as Russia, edit out the German troops in West Russia and Belorussia.
You will now see that not only can the armour blitz from Caucus over West Russia to Belorussia, but so can the infantry. But not the artillery. This is because of "isInfantry"
To recreate an example related to air transportables:
Go back into the POS2 xml and change Russia's techattachment paratroopers to "true". Both infantry and artillery are already isairtransportable
Launch the game as above.
Now not only can the infantry blitz, but the artillery can also.
So any unit with 2+ movement can always blitz, if it has isinfantry, or if it has isairtransportable and that nation has the paratrooper tech. This is regardless of the canBlitz unit attachment, and regardless of whether terrain restrictions should prevent blitz (which can be seen by heroes blitzing all terrains in Greyhawk Wars).
```

**Comment by DanVanAtta** [comment]
There we go, yeah, had to be POS2, not POS.

**Comment by DanVanAtta** [comment]
The problem runs pretty deep in the code. There is no good way to know if a unit is being transported, and so there are logic hacks to allow paratroop blitzing.

Paratroops have a special type of blitz, if you paradrop and capture a territory, then you can paradrop on territories behind that territory. A side effect of that rule, any unit that can blitz in that manner, when given more than 1 movement, will be able to blitz.

**Comment by panguitch** [comment]
> There is no good way to know if a unit is being transported
Yes, this is probably the root cause of the other paratrooper issue I've logged as well: https://github.com/triplea-game/triplea/issues/445

**Comment by Cernelius** [comment]
> Paratroops have a special type of blitz, if you paradrop and capture a territory, then you can paradrop on territories behind that territory. A side effect of that rule, any unit that can blitz in that manner, when given more than 1 movement, will be able to blitz.

Actually, this is **a second bug** you found. Paratrooping movement can only be extended by blitzing, not by other paratroopers (almost sure; if anyone can please confirm?). And with blitzing I mean that the paratroopers can go through all territories the blitzed unit has taken before the one it ended the movement into, as the last one is not blitzed, but normally captured at end combat movement (in the basic games, it is only one but, say, if you have a blitzing unit that moves 3, in a custom game, then literal transposition of the rule should allow you to move your paratrooper through the first two blitzed one, as long as the blitzing unit ended its combat movement into a third (not blitzed, no matter if empty) one). So, here I've tested we have **a third bug** too, as I see that if an armour takes 2 empty territories, then the paratroopers can go over both, while they should be allowed to go over the first one only, as it is the only one being actually blitzed.

I've googled something to confirm this:
https://boardgamegeek.com/thread/363104/aa50-rulebook-pdf-and-faq-now-available

Looks like the whole paratrooping + blitzing thing is a bit of a mess... I guess whoever coded them just expected the players to know the rules and limit themselves?

I'll let the developers decide if they think better to keep these other couple bugs here or make one or two more issues about those.

**Comment by Cernelius** [comment]
A question, at this point, is that, if I have 1 armour and I move into an undefended territory with it, would this be enough for the bomber to move through that territory, or do I need that armour also to do its second movement, anywhere?
For example:
1) I have 1 armour that moves from territory A to an undefended hostile territory B and ends combat move there, not using its second movement; can a bomber paratroop through territory B into a hostile territory C?
2) I have 1 armour that moves from territory A to an undefended hostile territory B and then moves back to territory A (or to whatever other friendly territory); can a bomber paratroop through territory B into a hostile territory C?
3) I have 1 armour that moves from territory A to an undefended hostile territory B and then moves into a hostile territory C; can a bomber paratroop through territory B into territory C?
4) I have 1 armour that moves from territory A to an undefended hostile territory B and then moves into a hostile territory C; can a bomber paratroop through territory B into a hostile territory D?

In my opinion, by rules, 1 should not be allowed, while 2, 3 and 4 should be allowed.
However, this would make no sense, in my mind, as if I can move through territory B if an armour takes it and keeps going on to whatever other territory, friend or foe, not having to be the same one my bomber ends up into and comprising going back whence it came (which is counted as a blitz move too), then I believe it would make the only sense that I can go through that territory also if the armour didn't do anything else but taking that territory (not using its second movement), even tho I don't believe this would rate as a blitz move.
On the other hand, if the armour would only take B with its second move (because it was 2 friendly territories far away), then I would agree that the bomber can't go through B.
So, basically, the rule would only make sense, in my mind, if the bomber is always allowed to go through undefended territories that a tank took on its first move only, no matter if then it does a second move or not. This would expand on custom maps for whatever move of a blitz-capable unit but the last one.
  @panther2  you seem to know the rules very well. Which ones of the above 4 points do you think should be legal and what not? Or maybe I'm missing some errata on this point? Personally, I would just prefer that the rule would be you can't ever paratroop past land territories that were hostile at the start of the turn, no matter blitzing, but this would be just my preference.

**Comment by Cernelius** [comment]
Also, a brief off topic, not related to blitzing, while we are on the paratrooper stuff: the engine behaviour is that if a territory has an AA gun only, in v3, it gets taken by paratroopers without the AA gun being able to fire on the paratrooping bombers. Thus 1 bomber paratrooping 1 infantry takes 1 territory with 1 AA gun only with no risk of being hit by AA fire. Can anyone confirm this is correct?
(I think it is, because AA guns alone can't start a battle, but I'm not 100% sure)

**Comment by Cernelius** [comment]
Basically, just asking to be confirmed that this one:

https://boardgamegeek.com/thread/363104/aa50-rulebook-pdf-and-faq-now-available
_Q. Bombers carrying Paratroopers must stop moving in the first hostile territory they enter. If a tank is blitzing through an unoccupied hostile territory, does a bomber entering that territory during the same Combat Movement phase have to stop there, or can it keep moving?
A. Since the territory is captured as soon as the blitzing tank enters it, the territory is considered friendly at that point, and the bomber may continue its movement._

was just an old FAQ and it is not true anymore (thus nevermind what I said; the problem would be just that the land territory only and always needs to be not hostile at start turn), as I see in the current official errata.
I should not have stopped searching at the first "official" explanation I googled, without double checking elsewhere if anything changed.

**Comment by panther2** [comment]
I am sorry, @Cernelius , for any reason I missed this issue and saw it only by now.

However, this is the from the latest AA50-FAQ (from September 3, 2014) that had been officially published:

Q. Bombers carrying Paratroopers must stop moving in the first hostile territory they enter. If a
tank is blitzing through an unoccupied hostile territory, does a bomber entering that territory
during the same Combat Movement phase have to stop there, or can it keep moving?
A. It must stop. Even though the territory is captured as soon as the blitzing tank enters it, and the
territory is considered friendly at that point, all combat movement is simultaneous. This means that
the bomber and tank entered the territory at the same time, when it was still hostile.

So yes, the PDF from your source is outdated and thus obsolete.
